src/main/kotlin/nl/connectplay/scoreplay/data/DatabaseGameRepository.kt

```diff
@@ -41,16 +44,131 @@ class DatabaseGameRepository(private val database: Database) : GameRepository {
+                                    // getInt() returns 0 if value is SQL NULL
+                                    minPlayers = rs.getInt("minimum_player_count").let { if (it > 0 ) it else null },
+                                    maxPlayers = rs.getInt("maximum_player_count").let { if (it > 0 ) it else null },
+                                    duration = rs.getInt("duration") .let { if (it > 0 ) it else null },
+                                    minAge = rs.getInt("minimum_age").let { if (it > 0 ) it else null },
                                     releaseDate = rs.getDate("release_date")?.toLocalDate()?.toKotlinLocalDate(),
                                 )
                             )
                         }
-                        list.toList() // can be empty
+                        list.toList() // Can be empty
+                    }
+                }
+            }
+        }.await()
+    }
+
+    override suspend fun addGame(create: CreateGameDto): GameDto = coroutineScope {
+        async {
+            // SQL Statement to db
+            database.connection?.use { conn ->
+                val sql = """
+                    INSERT INTO games
+                      (name, description, publisher, minimum_player_count, maximum_player_count, duration, minimum_age, release_date)
+                    VALUES (?, ?, ?, ?, ?, ?, ?, ?)
+                """.trimIndent()
+
+                conn.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS).use { stmt ->
+                    stmt.setString(1, create.name)
+                    stmt.setString(2, create.description)
+                    stmt.setString(3, create.publisher)
+                    if (create.minPlayers != null) stmt.setInt(4, create.minPlayers) else stmt.setNull(4, java.sql.Types.INTEGER)
+                    if (create.maxPlayers != null) stmt.setInt(5, create.maxPlayers) else stmt.setNull(5, java.sql.Types.INTEGER)
+                    if (create.duration != null) stmt.setInt(6, create.duration) else stmt.setNull(6, java.sql.Types.INTEGER)
+                    if (create.minAge != null) stmt.setInt(7, create.minAge) else stmt.setNull(7, java.sql.Types.INTEGER)
+
+                    if (create.releaseDate != null) {
+                        // Convert kotlinx.datetime.LocalDate -> java.time.LocalDate -> java.sql.Date
+                        val javaLocal = java.time.LocalDate.of(create.releaseDate.year, create.releaseDate.month, create.releaseDate.day)
+                        stmt.setDate(8, Date.valueOf(javaLocal))
+                    } else {
+                        stmt.setNull(8, java.sql.Types.DATE)
+                    }
+
+                    stmt.executeUpdate()
+
+                    // Read generated id
+                    val rs = stmt.generatedKeys
+                    val generatedId = if (rs.next()) rs.getInt(1) else throw IllegalStateException("Failed to retrieve generated id")
```

Comment:
je zou dit misschien iets cleaner kunnen doen met een RETURNING clausule in je query, maar dit is ook goed